automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/BatchNodepoolCreationSpec.scala

```diff
@@ -0,0 +1,171 @@
+
+          _ = monitorCreationResult.map(_.getNodePoolsList().size()) shouldBe Some(
+            defaultBatchNodepoolRequest.numNodepools.value + 1
+          )
+        } yield ()
+      }
+
+      test.unsafeRunSync()
+    }
+  }
+
+  "app creation with batch nodepool creation should work" - {
+    withNewProject { googleProject =>
+      val test = LeonardoApiClient.client.use { c =>
+        implicit val httpClient = c
+
+        val appDoneCheckable: DoneCheckable[GetAppResponse] =
+          x => x.status == AppStatus.Running
+
+        val appDeletedDoneCheckable: DoneCheckable[List[ListAppResponse]] =
+          x => x.map(_.status).distinct == List(AppStatus.Deleted)
+
+        val appName1 = AppName("app1")
+        val app1DeletedDoneCheckable: DoneCheckable[List[ListAppResponse]] =
+          x => x.filter(_.appName == appName1).map(_.status).distinct == List(AppStatus.Deleted)
+
+        for {
+          clusterName <- IO.fromEither(KubernetesNameUtils.getUniqueName(KubernetesClusterName.apply))
+          _ <- LeonardoApiClient.batchNodepoolCreate(
+            googleProject,
+            defaultBatchNodepoolRequest.copy(clusterName = Some(clusterName), numNodepools = NumNodepools(1))
+          )
+          getCluster = gkeServiceResource.use { gkeClient =>
+            val id = KubernetesClusterId(googleProject, LeonardoConfig.Leonardo.location, clusterName)
+            gkeClient.getCluster(id)
+          }
+
+          monitorBatchCreationResult <- testTimer.sleep(30 seconds) >> streamFUntilDone(getCluster, 20, 10 seconds)(
+            testTimer,
+            clusterDoneCheckable
+          ).compile.lastOrError
+
+          //here we sleep, because the above verifies the google state and we need to wait until leo has polled and updated its internal state to proceed
+          //it is a long time because there is a lot of stuff leo has to do besides GKE entity creation
+          _ <- testTimer.sleep(5 minutes)
+
+          _ = monitorBatchCreationResult.map(_.getNodePoolsList().size()) shouldBe Some(2)
+
+          diskConfig1 = Some(PersistentDiskRequest(DiskName("disk1"), None, None, Map.empty))
+
+          _ <- loggerIO.info("About to create app1")
+
+          _ <- LeonardoApiClient.createApp(googleProject,
+                                           appName1,
+                                           createAppRequest = defaultCreateAppRequest.copy(diskConfig = diskConfig1))
+
+          _ <- loggerIO.info("About to get app1")
+
+          getApp1 = LeonardoApiClient.getApp(googleProject, appName1)
+          monitorApp1CreationResult <- testTimer.sleep(30 seconds) >> streamFUntilDone(getApp1, 60, 10 seconds)(
+            testTimer,
+            appDoneCheckable
+          ).compile.lastOrError
+
+          _ <- loggerIO.info(s"app1 monitor result: ${monitorApp1CreationResult}")
+
+          clusterAfterApp1 <- getCluster
+          _ = clusterAfterApp1.map(_.getNodePoolsList().size()) shouldBe Some(2)
```

Comment:
What do you think about also adding a check that hits the proxy URL and verifies an OK http response?

Reply:
hmm... we could do this, but don't we do that in the monitoring code itself?